src/serializer/ResidualFunctions.js

```diff
@@ -485,11 +489,13 @@ export class ResidualFunctions {
           let funcNode;
           let firstUsage = this.firstFunctionUsages.get(functionValue);
           invariant(insertionPoint !== undefined);
+          let hasFunctionArg = flatArgs.find(arg => this.generatedFunctionIds.has(arg)) !== undefined;
```

Comment:
Instead of reasoning over Babel nodes here, check if any `residualBinding.value instanceof FunctionValue`. Look at the `let flatArgs` statement above.